client/src/main/java/io/pravega/client/netty/impl/ClientConnectionInboundHandler.java

```diff
@@ -128,7 +128,8 @@ public void sendAsync(List<Append> appends, CompletedCallback callback) {
         promise.addListener(new GenericFutureListener<Future<? super Void>>() {
             @Override
             public void operationComplete(Future<? super Void> future) throws Exception {
-                callback.complete(new ConnectionFailedException(future.cause()));
+                Throwable cause = future.cause();
+                callback.complete(cause == null ? null : new ConnectionFailedException(cause));
```

Comment:
I'm not sure I understand how this was working at all, it looks like we were always completing with an exception (which is now fixed here).

Following the code flow, we get here from `appendSetup`, so I'm not sure why `appendSetup` wasn't always failing.